### PR TITLE
8288623: Move Continuation classes out of javaClasses.hpp

### PR DESCRIPTION
**Pull request by iklam**
javaClasses.hpp is getting too big - it contains the C++ representation of over 50 Java classes.

The RFE moves the following classes into a new file, continuationJavaClasses.hpp. The naming follows the same pattern as the existing header share/jvmci/jvmciJavaClasses.hpp.

- jdk_internal_vm_ContinuationScope
- jdk_internal_vm_Continuation
- jdk_internal_vm_StackChunk

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288623](https://bugs.openjdk.org/browse/JDK-8288623): Move Continuation classes out of javaClasses.hpp

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [a2179280](https://git.openjdk.org/jdk/pull/9191/files/a217928085ddd2c047a33ab3f856370fe6c1bb31)
 * @kristylee88 (no known github.com user name / role) ⚠️ Review applies to [ebc3f9f8](https://git.openjdk.org/jdk/pull/9191/files/ebc3f9f8caeb657bd3d9a1614d5b91d5d639b333)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [a2179280](https://git.openjdk.org/jdk/pull/9191/files/a217928085ddd2c047a33ab3f856370fe6c1bb31)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9191/head:pull/9191` \
`$ git checkout pull/9191`

Update a local copy of the PR: \
`$ git checkout pull/9191` \
`$ git pull https://git.openjdk.org/jdk pull/9191/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9191`

View PR using the GUI difftool: \
`$ git pr show -t 9191`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9191.diff">https://git.openjdk.org/jdk/pull/9191.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/classfile/continuationJavaClasses.inline.hpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_CLASSFILE_JDK_INTERNAL_VM_STACKCHUNK_INLINE_HPP
+#define SHARE_CLASSFILE_JDK_INTERNAL_VM_STACKCHUNK_INLINE_HPP
```
Comment (stefank) [inline src/hotspot/share/classfile/continuationJavaClasses.inline.hpp:26]: This include guard doesn't follow the current naming convention.

**Review by stefank — COMMENTED** [review]
If we are about to start to squirrel away parts of javaClasses into new files, I wonder if the discoverability of these functions will be better if we use this naming instead:
```
javaClasses.hpp
javaClassesContinuation.hpp
javaClassesInvoke.hpp
...
```
instead of:
```
javaClasses.hpp
continuationJavaClasses.hpp
invokeJavaClasses.hpp
...
```

**Comment by coleenp** [comment]
The naming of this follows what jvmci did - they have jvmciJavaClasses.hpp/cpp.  Discovering continuation* is also something useful.

**Comment by iklam** [comment]
> If we are about to start to squirrel away parts of javaClasses into new files, I wonder if the discoverability of these functions will be better if we use this naming instead:
> 
> ```
> javaClasses.hpp
> javaClassesContinuation.hpp
> javaClassesInvoke.hpp
> ...
> ```
> 
> instead of:
> 
> ```
> javaClasses.hpp
> continuationJavaClasses.hpp
> invokeJavaClasses.hpp
> ...
> ```

I am not in love with either naming convention (the names are too long), but the latter sounds more grammatically correct. Also, as Coleen mentioned, if we choose the former, we need to rename jvmciJavaClasses.hpp as well.

The former can be discovered as `javaClasses*` and the latter as `*avaClasses*`, so there's not much difference.

**Comment by stefank** [comment]
I've talked to Ioi and Coleen about whether `javaClasses` or `continuations` should be the considered the most relevant subsystem. Their take is that `continuations` is. I'm warming up to that thought, but I think I'd then would prefer if continuationsJavaClasses.hpp etc. get moved to where the other continuation files are, in the runtime/ directory.

**Review by kristylee88 — APPROVED** [review]

**Review by stefank — CHANGES_REQUESTED** [review]
I started to add comments to the PR, but then realized it would be easier to just provide a patch with my suggestions. Please take a look at:
https://github.com/stefank/jdk/tree/pr_9191

**Comment by iklam** [comment]
Hi @stefank, thank you so much for providing the additional fixes. I have incorporated them into my PR and also synced with Coleen's javaThread.hpp split PR (#9218). Doing more thorough build testing now.

**Review by stefank — APPROVED** [review]

**Review by stefank — APPROVED** [review]

**Review by coleenp — APPROVED** [review]

**Comment by iklam** [comment]
Thanks to @stefank and @coleenp for the review.

- Passes our CI tier1 and build-tier5.
- GHA linux-x86 errors are unrelated to my changes.

/integrate

**Pull request closed without merging**